source/OptimaJet.DWKit.StarterApplication/Services/Workflow/WorkflowProductService.cs

```diff
@@ -133,29 +128,43 @@ public async Task ProductProcessChangedAsync(ProductProcessChangedArgs args)
                  .Include(p => p.Project)
                  .ThenInclude(pr => pr.Owner)
                  .FirstOrDefaultAsync();
+
             if (product == null)
             {
                 Log.Error($"Could find Product for ProcessId={args.ProcessId}");
                 return;
             }
 
-            await RemoveTasksByProductId(product.Id);
+            var tasks = await ReassignUserTasksForProduct(product);
+
+            await SendNotifications(tasks, product, args);
```

Comment:
The `SendNotifications` should be happening inside of `ReassignUserTasksForProduct` since notification also needs to happen for your new situation of reassigning the project to a different user.

Reply:
ok, I was actually wondering about that and forgot to ask if we should spam notifications when this happens :p